### PR TITLE
Fix Maven plugin validation warnings

### LINKED ISSUE #22487: Plugin validation issues exist with 2 plugins
##### **Overview of the issue**

When I create a new project with `blog-oauth2.jdl` (after fixing for v8), there's some warnings about Maven plugins:

```
[WARNING] Plugin validation issues were detected in 2 plugin(s)
[WARNING]
[WARNING]  * com.github.eirslett:frontend-maven-plugin:1.12.1
[WARNING]  * org.sonarsource.scanner.maven:sonar-maven-plugin:3.9.1.2184
[WARNING]
[WARNING] For more or less details, use 'maven.plugin.validation' property with one of the values (case insensitive): [BRIEF, DEFAULT, VERBOSE]
```

##### **Motivation for or Use Case**

It doesn't look good to have warnings from build files.

##### **Reproduce the error**

```
jhipster jdl blog-oauth2.jdl
# fix JDL by adding User with builtInEntity and run again
```

These warnings happen as part of the app creation process.

##### **JHipster Version(s)**

`main` branch

```
blog@0.0.1-SNAPSHOT /Users/mraible/Downloads/oauth
└── (empty)
```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "applicationIndex": 0,
  "applicationType": "monolith",
  "authenticationType": "oauth2",
  "baseName": "blog",
  "buildTool": "maven",
  "cacheProvider": "ehcache",
  "clientFramework": "angular",
  "clientTheme": "none",
  "clientThemeVariant": "",
  "creationTimestamp": 1686436859339,
  "databaseType": "sql",
  "devDatabaseType": "h2Disk",
  "devServerPort": 4200,
  "dtoSuffix": "DTO",
  "enableGradleEnterprise": false,
  "enableHibernateCache": true,
  "enableSwaggerCodegen": false,
  "enableTranslation": true,
  "entities": [
    "Blog",
    "Post",
    "Tag"
  ],
  "entitySuffix": "",
  "jhiPrefix": "jhi",
  "jhipsterVersion": "7.10.0",
  "languages": [
    "en",
    "fr"
  ],
  "lastLiquibaseTimestamp": 1686436861000,
  "messageBroker": "no",
  "nativeLanguage": "en",
  "packageFolder": "com/jhipster/demo/blog",
  "packageName": "com.jhipster.demo.blog",
  "pages": [],
  "prodDatabaseType": "mysql",
  "reactive": false,
  "searchEngine": "no",
  "serverPort": 8080,
  "serviceDiscoveryType": "no",
  "skipCheckLengthOfIdentifier": false,
  "skipClient": false,
  "skipFakeData": false,
  "skipUserManagement": true,
  "testFrameworks": [
    "cypress"
  ],
  "websocket": "no",
  "withAdminUi": true
}
</pre>
</details>

##### **Environment and Tools**

openjdk version "17.0.5" 2022-10-18 LTS
OpenJDK Runtime Environment GraalVM 22.3.0 (build 17.0.5+8-LTS)
OpenJDK 64-Bit Server VM GraalVM 22.3.0 (build 17.0.5+8-LTS, mixed mode, sharing)

git version 2.39.2 (Apple Git-143)

node: v18.13.0
npm: 9.6.5

Docker version 23.0.5, build bc4487a

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Blog {
  name String required minlength(3)
  handle String required minlength(2)
}
entity Post {
  title String required
  content TextBlob required
  date Instant required
}
entity Tag {
  name String required minlength(2)
}
relationship ManyToOne {
  Blog{user(login)} to User with builtInEntity
  Post{blog(name)} to Blog
}
relationship ManyToMany {
  Post{tag(name)} to Tag{entry}
}

search Blog, Post, Tag with no
paginate Post, Tag with infinite-scroll

</pre>
</details>

### PR DESCRIPTION
**Pull request by mraible**
Fixes #22487. 

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/templates/pom.xml.ejs [hunk 1]
```diff
@@ -756,38 +756,13 @@
...
-                            <goals>
-                                <goal>copy-resources</goal>
-                            </goals>
-                            <configuration>
-                                <outputDirectory>${project.build.directory}/classes</outputDirectory>
-                                <useDefaultDelimiters>false</useDefaultDelimiters>
-                                <delimiters>
-                                    <delimiter>#</delimiter>
-                                </delimiters>
-                                <resources>
```
Comment (aheritier) [inline generators/server/templates/pom.xml.ejs:772]: Maybe you will want to keep a resources definition in your POM.

Maven by default uses : 

```
    <resources>
      <resource>
        <directory>${project.basedir}/src/main/resources</directory>
      </resource>
    </resources>
```

https://maven.apache.org/ref/3.9.5/maven-model-builder/super-pom.html

In Spring Boot, they modified it to filter application yml/properties files

```
<resources>
<resource>
<directory>${basedir}/src/main/resources</directory>
<filtering>true</filtering>
<includes>
<include>**/application*.yml</include>
<include>**/application*.yaml</include>
<include>**/application*.properties</include>
</includes>
</resource>
<resource>
<directory>${basedir}/src/main/resources</directory>
<excludes>
<exclude>**/application*.yml</exclude>
<exclude>**/application*.yaml</exclude>
<exclude>**/application*.properties</exclude>
</excludes>
</resource>
</resources>
```

https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-starter-parent/3.1.5/spring-boot-starter-parent-3.1.5.pom

And AFAIU in JHipster you were filtering YML files under the config folder

I suppose you want to continue to filter these files to keep the compatibility

Comment (mraible) [reply generators/server/templates/pom.xml.ejs:772]: Can you please make a suggested change so I know what you mean?

Comment (aheritier) [reply generators/server/templates/pom.xml.ejs:772]: @mraible I did another PR, it was easier to do it from my IDE: #23946

I found something which should remove the warning while keeping exactly the same behavior than today. 

(I sadly cannot test it myself easily right now)

**Comment by mraible** [comment]
Closing in favor of #23946.

**Pull request closed without merging**